pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -263,6 +264,18 @@ public class Utils {
         return nodes
     }
 
+    static void markExecutedStagesOnAction(CpsScript script, String astUUID) throws Exception {
```

Comment:
Seems kind of hacky.

Reply:
It is a bit hacky, yes, but given that I don't actually know which `pipeline` block (if there are multiple being read from the `Jenkinsfile` and shared libraries) is going to be executed until it actually *is* executed...and hey, it works. If a better idea comes along in the future, I'll take advantage of it. =)